src/radar_control.py

```diff
@@ -1161,29 +1117,18 @@ def main():
                 time_remains = True
 
                 while time_remains:
-                    if ave_params.num_sequences == 0 and aveperiod.cfs_flag:
+                    if ave_params.num_sequences == 0 and isinstance(
+                        aveperiod, CFSAveragingPeriod
+                    ):
                         cfs_time = time.time()
-                        updated = cfs_block(
+                        cfs_block(
                             ave_params,
-                            cfs_params_dict,
                             [
                                 radctrl_inproc_socket,
                                 radctrl_brian_socket,
                                 driver_comms_socket,
                             ],
                         )
-                        if not updated:
-                            # Need to update current beam frequency to pre-determined value
-                            beam = aveperiod.beam_iter
-                            cfs_set_freqs = dict()
-                            for slice_id in cfs_params_dict[aveperiod].beam_frequency[
-                                beam
-                            ]:
-                                cfs_set_freqs[slice_id] = cfs_params_dict[
-                                    aveperiod
-                                ].beam_frequency[beam][slice_id]
-                            aveperiod.update_cfs_freqs(cfs_set_freqs)
-
```

Comment:
This block was needed because in the event that CFS does not need to change the frequency  during this integration time, we still need to change what frequency the N200 is transmitting  since the frequency changes per beam. I may have missed if `update_cfs_freqs` was called elsewhere in the code but it needs to be run after the cfs block if the cfs block doesn't pick a new freq.

Maybe we should not have `select_cfs_freq` in the aveperiod run `update_cfs_freqs` at all and just always run it after the cfs block regardless.

Reply:
Ah okay, thanks for that context. I've put the update_cfs_freqs back in, so it runs after cfs_block regardless of whether a new search was conducted or not.